snuba/utils/streams/consumers/backends/kafka.py

```diff
@@ -275,6 +275,38 @@ def seek(self, offsets: Mapping[TopicPartition, int]) -> None:
 
         self.__seek(offsets)
 
+    def pause(self, streams: Sequence[TopicPartition]) -> None:
+        if self.__state in {KafkaConsumerState.CLOSED, KafkaConsumerState.ERROR}:
+            raise InvalidState(self.__state)
+
+        self.__consumer.pause(
+            [
+                ConfluentTopicPartition(stream.topic, stream.partition)
+                for stream in streams
+            ]
+        )
+
+        # XXX: Seeking to a specific partition offset and immediately pausing
+        # that partition causes the seek to be ignored for some reason.
+        self.seek(
+            {
+                stream: offset
+                for stream, offset in self.__offsets.items()
+                if stream in streams
+            }
+        )
```

Comment:
I should probably open up an issue on confluent-kafka about this.

Reply:
well, indeed,  this is gross if it is not a bug.

Reply:
I can't imagine a situation where this is an expected behavior.